test/evp_libctx_test.c

```diff
@@ -263,6 +263,77 @@ static int test_cipher_reinit(int test_id)
     return ret;
 }
 
+/*
+ * This test only uses a partial block (8 bytes) of input for each
+ * EVP_EncryptUpdate() in order to test that the second init/update is not using
+ * a leftover buffer from the first init/update.
+ * Note: some ciphers don't need a full 16 byte block to produce output.
+ */
+static int test_cipher_reinit_partialupdate(int test_id)
+{
+    int ret = 0, out1_len = 0, out2_len = 0;
+    EVP_CIPHER *cipher = NULL;
+    EVP_CIPHER_CTX *ctx = NULL;
+    unsigned char out1[256];
+    unsigned char out2[256];
+    static const unsigned char in[8] = {
```

Comment:
What about ciphers with a 64 bit block size? This won't end up being a partial block in those cases.

Reply:
I thought of this when I had a quick look but didn't comment.

What about adding in increments of 3 and 5 bytes alternately?  That would handle all multiples of 8 naturally and be easy enough to code: `length = 8 - length ` or `length ^= 6`.

Reply:
That change picked up des/tdes being broken also..